ods_ci/tests/Tests/100__deploy/130__operators/140__serverless_operator/140__serverless_operator.robot

```diff
@@ -0,0 +1,95 @@
+Suite Setup      Suite Setup
+Suite Teardown   Suite Teardown
+
+
+*** Variables ***
+${SERVERLESS_APPNAME}  Red Hat OpenShift Serverless
+${SERVERLESS_OPERATOR_NAME}    Red Hat OpenShift Serverless
+${KNATIVESERVING_NS}    knative-serving
+${ISTIO_NS}     istio-system
+
+
+*** Test Cases ***
+Validate DSC creates all Serverless CRs
+    [Documentation]  The purpose of this Test Case is to validate the creation
+    ...    of Serverless Custom Resources
+    [Tags]  Operator    ODS-2600
+    Assign Vars According To Product    ${PRODUCT}
+    Check And Install Operator in Openshift    ${SERVERLESS_APPNAME}    ${SERVERLESS_OPERATOR_NAME}
+    Check And Install Operator in Openshift    ${OPERATOR_APPNAME}    ${OPERATOR_NAME}
+    Is Resource Present     KnativeServing    knative-serving     ${KNATIVESERVING_NS}
+    Check Status     oc get KnativeServing knative-serving -n ${KNATIVESERVING_NS} -o json | jq '.status.conditions[] | select(.type=="Ready") | .status'     KnativeServing    "True"    # robocop: disable
+    Is Resource Present     Gateway    knative-ingress-gateway     ${KNATIVESERVING_NS}
+    Is Resource Present     Gateway    knative-local-gateway     ${KNATIVESERVING_NS}
+    Is Resource Present     Service    knative-local-gateway     ${ISTIO_NS}
+    Is Resource Present     deployment    controller     ${KNATIVESERVING_NS}
+    Wait For Pods Numbers  2    namespace=${KNATIVESERVING_NS}
+    ...    label_selector=app.kubernetes.io/component=controller    timeout=120
+    ${pod_names}=    Get Pod Names    ${KNATIVESERVING_NS}    app.kubernetes.io/component=controller
+    Verify Containers Have Zero Restarts    ${pod_names}    ${KNATIVESERVING_NS}
+    ${podname}=    Get Pod Name   ${OPERATOR_NAMESPACE}    label_selector=name=rhods-operator
+    Check For Errors On Operator Logs    ${podname}    ${OPERATOR_NAMESPACE}
+    Read DSC Conditions    ${KNATIVESERVING_NS}    default-dsc
+
+
+*** Keywords ***
+Suite Setup
+    [Documentation]    Suite Setup
+    Set Library Search Order    SeleniumLibrary
+    RHOSi Setup
+
+Suite Teardown
+    [Documentation]    Suite Teardown
+    Close All Browsers
+    RHOSi Teardown
+
+Assign Vars According To Product
+    [Documentation]    Assign vars related to product
+    [Arguments]    ${product}
+    IF    "${product}" == "RHODS"
+        Set Suite Variable    ${OPERATOR_APPNAME}     Red Hat OpenShift AI
+        Set Suite Variable    ${OPERATOR_NAME}    Red Hat OpenShift AI
+    ELSE IF    "${product}" == "ODH"
+        Set Suite Variable    ${OPERATOR_APPNAME}  Open Data Hub Operator
+        Set Suite Variable    ${OPERATOR_NAME}    Open Data Hub Operator
+    END
+
+Is Resource Present
+    [Documentation]    Check CR
+    [Arguments]       ${resource}     ${resource_name}    ${namespace}
+    ${rc}=     Run and Return Rc
+    ...  oc get ${resource} ${resource_name} -n ${namespace}
+    Should Be Equal    "${rc}"    "0"    msg=${resource} does not exist
+
+Check Status
```

Comment:
maybe this keyword could go in a common resource file so it can be re-used in different suites

`Resource Status Should Be` could be more RF-style name

Reply:
changed name and moved to `ods_ci/tests/Resources/OCP.resource`

Reply:
still see original version

Reply:
just committed